smoke_tests/standalone_smoke_test.py

```diff
@@ -0,0 +1,324 @@
+# Copyright 2014-2020 Scalyr Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.K
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+
+import datetime
+import time
+import json
+import re
+
+import pytest  # type: ignore
+
+from scalyr_agent.__scalyr__ import PACKAGE_INSTALL, DEV_INSTALL
+from scalyr_agent import compat
+
+from smoke_tests.tools.agent_runner import AgentRunner
+from smoke_tests.tools.request import ScalyrRequest
+
+from six.moves import range
+import six
+
+_TIMESTAMP_PATTERN = r"\d{4}-\d{2}-\d{2}\s+\d{2}:\d{2}:\d{2}.\d+Z"
+_LEVEL_PATTERN = r"INFO|WARNING|ERROR|DEBUG"
+_LOGGER_NAME_PATTERN = r"[^\]]*"
+_FILE_NAME_PATTERN = r"[^\]:]*"
+
+_AGENT_LOG_LINE_BASE_PATTERN = r"\s*(?P<timestamp>{0})\s+(?P<level>{1})\s+\[(?P<logger>{2})\]\s+\[(?P<file>{3}):\d+\]\s+"
+
+
+def _make_agent_log_line_pattern(
+    timestamp=_TIMESTAMP_PATTERN,
+    level=_LEVEL_PATTERN,
+    logger_name=_LOGGER_NAME_PATTERN,
+    file_name=_FILE_NAME_PATTERN,
+    message=None,
+):
+    """
+    Build regex pattern for 'agent.log' log lines.
+    """
+    base_pattern = _AGENT_LOG_LINE_BASE_PATTERN.format(
+        timestamp, level, logger_name, file_name
+    )
+    if message:
+        pattern_str = "{0}{1}".format(base_pattern, message)
+    else:
+        pattern_str = base_pattern
+    pattern = re.compile(pattern_str)
+
+    return pattern
+
+
+class AgentVerifier(object):
```

Comment:
We could also signal that this is a base class by using ``@six.add_metaclass(abc.ABCMeta)`` decorator which works under Python 2 and Python 3.

We just need to confirm if it works under Python 2.6 (I know that works under 2.7, but not sure about 2.6.).